pre_commit/languages/docker.py

```diff
@@ -76,7 +78,28 @@ def install_environment(
         os.mkdir(directory)
 
 
+def _docker_is_rootless() -> bool:
+    returncode, stdout, stderr = cmd_output(
+        'docker', 'system', 'info',
+    )
+    for line in stdout.splitlines():
+        # rootless docker has "rootless"
+        # rootless podman has "rootless: true"
+        if line.strip().startswith('rootless'):
+            if 'false' not in line:
+                return True
+            break
+    return False
+
+
+@functools.lru_cache(maxsize=1)
+def docker_is_rootless() -> bool:
```

Comment:
This should probably have a PEP257-compliant docstring

Reply:
To be fair, non of the other functions in this file have docstrings.

Reply:
Fair enough. I guess it's just my personal habit kicking in before the linters :)